src/agent/reqwest-retry/src/lib.rs

```diff
@@ -195,17 +105,12 @@ mod test {
         let invalid_url = "http://127.0.0.1:81/test.txt";
         let resp = reqwest::Client::new()
             .get(invalid_url)
-            .send_retry(
-                Duration::from_secs(1),
-                Duration::from_secs(300),
-                2i32,
-                to_backoff_response,
-            )
+            .send_retry(Duration::from_millis(1), 3)
             .await;
 
         if let Err(err) = &resp {
             let as_text = format!("{}", err);
-            assert!(as_text.contains(MAX_RETRY_ERROR_MESSAGE));
+            assert!(as_text.contains("request failed after"));
```

Comment:
I know this was the existing logic but maybe we need to introduce a RetryError for this case.
not a blocking comment.

Reply:
The siggestion here is to have send_retry return a Result<Response, SendRetryError> and move the type of failure inside the SendRetryError. That way we don't have to do string contains

Reply:
That's reasonable, but let's do that as a follow-on PR.